### ISSUE TITLE
coin.m4 does not work on AIX native shell

### ISSUE THREAD

**Issue by svigerske** [coin-or-tools/BuildTools#88]
Issue created by migration from Trac.

Original creator: @andrea5w

Original creation time: 2011-06-14 17:56:21

Assignee: @svigerske

Version: 0.6

Hi,

I'm trying to configure Clp 1.13.1 on AIX. When I do not set CONFIG_SHELL=PATH/bash, the configure script of CoinUtils fails and terminates without writing configuration files, but the overall configuration still continues:

```
checking for COIN-OR package Sample... skipped check via pkg-config, redirect to fallback
checking for COIN-OR package Sample (fallback)... ./configure[34393]: projtoprocess=${projtoprocess/$proj/$projrequires}: bad substitution
configure: configuring in Osi
```

It appears that the offending part is around line 4225 in coin.m4:

```
      # add projrequires to the front of the list of projects that have to be processed next
      # at the same time, remove $proj from this list
      projtoprocess=${projtoprocess/$proj/$projrequires}
```

I don't know what this is supposed to do, so I cannot fix it.  But I suggest to avoid any non-standard shell tools.

Thanks,

Andreas

**Comment by svigerske** [comment]
Comment by @svigerske created at 2011-06-14 18:25:34

Hi,

it's supposed to substitute `$proj` by `$projrequires` in `$projtoprocess`.

I comitted a fix to BuildTools/trunk. Before rerunning autotools and merging around, can you test if it works on AIX now? E.g., use Osi/trunk and run the autotools on your checkout.

Stefan

**Comment by svigerske** [comment]
Comment by @svigerske created at 2011-06-14 18:25:34

Changing status from new to assigned.

**Comment by svigerske** [comment]
Comment by @andrea5w created at 2011-06-14 22:10:46

Thanks, Stefan!  This seems to have fixed it.  I ran run_autotools on trunk and now the configuration went through.

Cheers,

Andreas

**Comment by svigerske** [comment]
Comment by @svigerske created at 2011-06-15 12:09:55

Fixed in BuildTools/trunk, stable/0.6, and stable/0.7.

Reran autotools for projects in CoinAll/trunk and CoinAll/stable/1.5.

Can create new releases for projects in CoinAll 1.6 on request.

Stefan

**Comment by svigerske** [comment]
Comment by @svigerske created at 2011-06-15 12:09:55

Resolution: fixed
